### ISSUE TITLE
Active Directory multiples domains in a forest

### ISSUE THREAD

**Issue by devaraja666** [jakubgarfield/Bonobo-Git-Server#377]
i don't find a way to add users from others domains in a forest

**Comment by larshg** [comment]
@devaraja666 there has been some updates on multiple domains - if you can add all users to the same membergroup (if there is a trust between domains?) then it should work? Can you test 6.1?

I'm no expert in AD multidomain, so maybe I'm off track here.

**Comment by devaraja666** [comment]
Hi @larshg I tried the new version, i can see now the users information for the accounts in the forest that are in the MemberGroup, but now i cant add group that have space in teh name, in previous versions the group name could have spaces, but now if the group name have spaces the site wont load, it only shows a white background.

This is my configuration:
`    <add key="AuthenticationProvider" value="Windows" />
    <add key="ImportWindowsAuthUsersAsAdmin" value="false" />
    <add key="MembershipService" value="ActiveDirectory" />
    <add key="ActiveDirectoryBackendPath" value="~\App_Data\ADBackend" />
    <add key="ActiveDirectoryMemberGroupName" value="Git" />
    <add key="ActiveDirectoryTeamMapping" value="Developers='Group Developers, Testers=Group Testers" />
    <add key="ActiveDirectoryRoleMapping" value="Administrator=GitAdmins" />`

In the log that i'm including you can see that it gave an [Error] Exception caught in Global.asax1

I try to put the groups inside a single quotation mark but they came back sating that it Could not find principal group: 'Group Testers' because is looking for teh group name 'Group Testers' insted of Group Testers

If i used a group with no spaces, it detect the menbers and the site does load.

[log-20170421.txt](https://github.com/jakubgarfield/Bonobo-Git-Server/files/947397/log-20170421.txt)

**Comment by larshg** [comment]
Hi

Thanks for the response.

I'll try to investigate where the group names with spaces doesn't work :-)

**Comment by willdean** [comment]
New logging looks good ;-)

**Comment by larshg** [comment]
@devaraja666 I just tested with a group name with spaces and it find that group without problems. Can you double check the spelling of the names etc?

I'll try add some more logging to the /home/Diagnostics page to see if we can query some useful info.

**Comment by willdean** [comment]
@larshg If there's any way to get the AD servers to list all the groups (I guess that's the sort of thing you were thinking of adding to the home/diag page), then could we also dump all that to the log file if there's a problem finding a group?  So the log file says "I couldn't find your group but here are all the ones I could find"

**Comment by larshg** [comment]
Yes, I was thinking about doing something like that. I'll see what I can do :)

**Comment by larshg** [comment]
@devaraja666 Can you try out the artifact from this build when it is done:
https://ci.appveyor.com/project/jakubgarfield/bonobo-git-server/build/564/artifacts

And navigate to /home/dianostics and verify that you can find the group names that you want to use.

**Comment by willdean** [comment]
@devaraja666 To go to /home/diagnostics you have to be running a browser on the same machine as is serving Bonobo...

**Comment by devaraja666** [comment]
@larshg I found the problem, i had 2 users in the "Group Developers" that where not part of the "Git" group. a soon i adeed the user to the git group the site load correctly. 

This does not happened in the previoues version, that i why i did not notices it, after doing alot of testing, creating new group, i added a user that was not included in the "ActiveDirectoryMemberGroup" Git, to a "TeamMapping" group and i got the error.

that is why the system  throw an Error occured and caught in Global.asax - System.NullReferenceException: Object reference not set to an instance of an object.

Because is triying to match a user in the team that is not part of the "ActiveDirectoryMemberGroup".

It could be a good idea to add a condition to ignore a user that is not part of the member group, or to do a warining in the log to show the user.

Thank you so much for your help.


[diagnostics.txt](https://github.com/jakubgarfield/Bonobo-Git-Server/files/951395/diagnostics.txt)
[log-20170424.txt](https://github.com/jakubgarfield/Bonobo-Git-Server/files/951396/log-20170424.txt)

**Comment by larshg** [comment]
Thanks for reporting back - I'll try to investigate and figure out how to properly handle it :)
